File: facts.d/get_directives_details.php
#!/usr/local/zend/bin/php
<?php
error_reporting(E_ERROR);

$zs_api_config_file='/.zsapi.ini';
$zs_api_target='localadmin';
$zs_api_config=parse_ini_file($zs_api_config_file, true);

$api_key_name=$zs_api_config[$zs_api_target]['zskey'];
$api_key_hash=$zs_api_config[$zs_api_target]['zssecret'];

$get_directives_details="/usr/local/zend/bin/zs-client.sh configurationDirectivesList --target=localadmin --output-format=json > /tmp/ZendDirectivesList";

exec($get_directives_details , $directives_exec_output);

$textJson = file_get_contents('/tmp/ZendDirectivesList');

//die ($directives_exec_output);

//print_r ($directives_exec_output);

//die;

$arr = json_decode($textJson, true);

//print_r ($arr);
//var_dump ($arr['directive']);
//print_r ($directives_exec_output[2]);

foreach($arr['responseData']['directives'] as $app) {
    $name = $app['name'];

    foreach($app as $key => $value) {
        if(! is_array($value))
            echo 'zend_directive_' . $key . '_' . $name . '=' . $value . PHP_EOL;
    }
}

File: facts.d/get_directives_details.py
#!/usr/bin/python

import json
import os.path
import re
import subprocess

zs_api_config_file = '/.zsapi.ini'
zs_api_target = 'localadmin'

if os.path.isfile("/usr/local/zend/bin/zs-client.sh"):

  directives_details = subprocess.check_output(["/usr/local/zend/bin/zs-client.sh", "configurationDirectivesList", "--target=localadmin", "--output-format=json"])

  ## Strip the PHP notices from the json
  directives_details = re.sub("Notice:.*\n", "", directives_details)
  ## Strip the newlines from the json
  directives_details = re.sub("\n", "", directives_details)

  f1=open('/tmp/puppet_zend_directives.txt', 'w')
  f1.write(directives_details)

  arr = json.loads(directives_details)

  for directive in arr[u"responseData"][u"directives"]:
    name = directive["name"]
    for key, value in directive.iteritems():
      if value and not isinstance(value, list) and not isinstance(value,dict):
        print ('zend_directive_' + key + '_' + name + '=' + value)
